app/services/edge_cache/buster.rb

```diff
@@ -0,0 +1,115 @@
+module EdgeCache
+  class Buster
+    PROVIDERS = %w[fastly nginx].freeze
+
+    def initialize(path)
+      # TODO: (Vaidehi Joshi) - Right now, we are checking that nginx is
+      # available on every purge request/call to this buster service. If we are going
+      # to bust multiple paths, we should be able to check that nginx is
+      # available just once, and persist it on the class with @provider_available?.
+      # Then, we could allow for an array of @paths = [] to be passed in,
+      # and on single buster instance could bust multiple paths in order.
+
+      @path = path
+      @provider = set_provider
```

Comment:
What if we made providers their own classes, e.g. `EdgeCache::Bust::Fastly` (assuming we rename this as suggested in the other comment, otherwise `EdgeCache::Buster::Fastly`), which expose a single `call(path)` class method? Then we could change this to 

This could greatly simplify the local `call` method, I'll leave a related comment there too.

Reply:
Yeah, I like this idea. I'll tackle it when I tackle the remaining TODO comment!